src/layout/index.tsx

```diff
@@ -20,8 +20,28 @@ const Layout: React.FC<LayoutProps> = ({
 }) => {
   const footerContainer = useRef<HTMLElement>(null);
 
+  const [clickActive, setClickActive] = useState(false);
+  const burgerRef = useRef<HTMLDivElement>(null);
+
+  const handleClick = (e: Event) => {
+    if (burgerRef.current && !burgerRef.current.contains(e.target as Node)) {
+      setClickActive(false);
+    }
+  };
+
+  useEffect(() => {
```

Comment:
Thinking more about this, I think actually all this `handleClick` now with `nav` as 100% width and height all you need is an `onClick` listener on the nav element through React. That would render the useEffect obsolete.

Reply:
Also `burgerRef`

Reply:
Yes! made it into a onclick function instead!